api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -998,18 +999,21 @@ public void markDirty(ResourceType resourceType, long resourceId) {
   }
 
   @Override
-  public DataDictionaryEntry findDataDictionaryEntry(String fieldName, DbCdrVersion cdrVersion) {
-    List<DbDataDictionaryEntry> dataDictionaryEntries =
-        dataDictionaryEntryDao.findByFieldNameAndCdrVersion(fieldName, cdrVersion);
-
-    if (dataDictionaryEntries.isEmpty()) {
+  public DataDictionaryEntry findDataDictionaryEntry(
+      String fieldName, String domain, DbCdrVersion cdrVersion) {
+    CdrVersionContext.setCdrVersionNoCheckAuthDomain(cdrVersion);
```

Comment:
This is what controls which cloud CDR we query and needs to be the first thing set in the Controller, so I would move it up.  Also, we should be using the `CdrVersionService` like so -> `cdrVersionService.setCdrVersion(cdrVersionId);`

Reply:
Done, moved to controller